server/src/controllers/utils/aplus.ts

```diff
@@ -27,17 +26,41 @@ export const validateAplusCourseId = (aplusCourseId: string): number => {
   return result.data;
 };
 
+/**
+ * Validates that an A+ API token was provided and parses it from the request
+ * header.
+ *
+ * @throws ApiError(400)
+ */
+export const parseAplusToken = (req: Request): string => {
+  const auth = req.headers.authorization;
+  if (!auth) {
+    throw new ApiError('no A+ API token provided', HttpCode.BadRequest);
+  }
+
+  const authArray = auth.split(' ');
+  const result = z
+    .tuple([z.literal('Aplus-Token'), z.string().length(40)])
+    .safeParse(authArray);
+  if (!result.success) {
+    throw new ApiError(result.error.message, HttpCode.BadRequest);
+  }
+
+  return authArray[1];
+};
```

Comment:
Is it fine to use 400 here? As far as Grades is concerned, we only want to know what the token string is, A+ handles all the authentication, so I'm not sure if 401 and 403 are appropriate.

I also need to confirm whether the length of the token is always 40.

Reply:
I think 400 is the correct response code here